src/Exceptions/ResponseException.php

```diff
@@ -0,0 +1,29 @@
+<?php
+
+namespace Foxy\FoxyClient\Exceptions;
+
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class ResponseException extends RequestException
+{
+
+    public function __construct(
+        string $message,
+        RequestInterface $request,
+        ResponseInterface $response,
+        \Throwable $previous = null
+    ) {
+        parent::__construct($message, $request, $response, $previous);
+    }
+
+    public function hasResponse(): bool
+    {
+        return true;
```

Comment:
This doesn't feel right. Are we checking if we have some response or not? It will always return true no matter what. 
Can this be changed to return **$this->response !== null;** like in RequestException.php?

@brettflorio

Reply:
This is ResponseException which is required to have response, so no reason to check.